java/ChartboostMediationDemoJava/src/main/java/com/example/chartboost/mediation/sdk/demo/java/AdController.java

```diff
@@ -0,0 +1,296 @@
+                    fullscreenAds.put(placementName, chartboostMediationFullscreenAdLoadResult.getAd());
+                    correspondingShowButton.setEnabled(fullscreenAds.get(placementName) != null);
+                } else {
+                    uiLogsListener.add(String.format("Error loading ad, code: %s : %s", chartboostMediationFullscreenAdLoadResult.getError().getCode(), chartboostMediationFullscreenAdLoadResult.getError().getMessage()));
+                }
+                loadButton.setEnabled(true);
+            }
+        };
+    }
+
+    private ChartboostMediationFullscreenAdListener createFullscreenAdListener(
+            String placementName,
+            Button showButton,
+            UILogsListener uiLogsListener
+    ) {
+        return new ChartboostMediationFullscreenAdListener() {
+
+            @Override
+            public void onAdClicked(@NonNull ChartboostMediationFullscreenAd chartboostMediationFullscreenAd) {
+                uiLogsListener.add("Ad clicked");
+            }
+
+            @Override
+            public void onAdClosed(@NonNull ChartboostMediationFullscreenAd chartboostMediationFullscreenAd, @Nullable ChartboostMediationAdException e) {
+                uiLogsListener.add("Ad closed");
+                ChartboostMediationFullscreenAdQueue queue = getAdQueue(placementName);
+                if (shouldUseQueue) {
+                    showButton.setEnabled(queue.hasNextAd());
+                } else {
+                    showButton.setEnabled(false);
+                }
+            }
+
+            @Override
+            public void onAdRewarded(@NonNull ChartboostMediationFullscreenAd chartboostMediationFullscreenAd) {
+                uiLogsListener.add("Ad rewarded");
+            }
+
+            @Override
+            public void onAdImpressionRecorded(@NonNull ChartboostMediationFullscreenAd chartboostMediationFullscreenAd) {
+                uiLogsListener.add("Impression recorded");
+            }
+
+            @Override
+            public void onAdExpired(@NonNull ChartboostMediationFullscreenAd chartboostMediationFullscreenAd) {
+                uiLogsListener.add("Ad expired");
+            }
+        };
+    }
+
+    // To listen to ad queue events, you can use the ChartboostMediationFullscreenAdQueueListener
+    private ChartboostMediationFullscreenAdQueueListener createFullscreenAdQueueListener(
+            String placementName, Button showButton,
+            UILogsListener uiLogsListener) {
+        return new ChartboostMediationFullscreenAdQueueListener() {
+
+            @Override
+            public void onFullScreenAdQueueUpdated(@NonNull ChartboostMediationFullscreenAdQueue adQueue, @NonNull AdLoadResult result, int numberOfAdsReady) {
+                uiLogsListener.add("Fullscreen ad queue has been updated for placement " + placementName + ". Number of ads ready to show: " + numberOfAdsReady);
+                showButton.setEnabled(numberOfAdsReady > 0);
```

Comment:
like here: if you stop the queue, number of ads ready goes to 0 but if you're in the middle of a non-queue'd load and it finishes, this could cause this to be disabled.
Just having it all in once place is probably more confusing than it's worth.